src/segments/argocd_test.go

```diff
@@ -0,0 +1,233 @@
+package segments
+
+import (
+	"fmt"
+	"github.com/jandedobbeleer/oh-my-posh/src/mock"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/stretchr/testify/assert"
+	mock2 "github.com/stretchr/testify/mock"
+	"path"
+	"testing"
+)
+
+func TestArgocdGetConfigFromOpts(t *testing.T) {
+	configFile := "/Users/posh/.configFile/argocd/config"
+	cases := []struct {
+		Case          string
+		Opts          string
+		Expected      string
+		ExpectedError string
+	}{
+		{Case: "invalid flag in opts", Opts: "--invalid", ExpectedError: argocdInvalidFlag},
+		{Case: "another invalid flag in opts", Opts: "-invalid", ExpectedError: argocdInvalidFlag},
+		{Case: "no config in opts", Opts: "--grpc-web", Expected: ""},
+		{Case: "config in opts", Opts: fmt.Sprintf("--grpc-web --config %s --plaintext", configFile), Expected: configFile},
+	}
+
+	for _, tc := range cases {
+		env := new(mock.MockedEnvironment)
+		env.On("Getenv", argocdOptsEnv).Return(tc.Opts)
+
+		argocd := &Argocd{
+			env:   env,
+			props: properties.Map{},
+		}
+		if len(tc.ExpectedError) > 0 {
+			_, err := argocd.getConfigFromOpts()
+			assert.EqualError(t, err, tc.ExpectedError, tc.Case)
```

Comment:
can you do an early return here instead of if/else?

```suggestion
			assert.EqualError(t, err, tc.ExpectedError, tc.Case)
			continue
```

Reply:
updated with early return.